src/core/sync/mutex.d

```diff
@@ -372,6 +372,9 @@ unittest
     // by checking that locking is not possible. This assumes
     // that the underlying implementation is well behaved
     // and makes the object non-lockable upon destruction.
+    // For example, Bionic doesn't appear to do so, so this test is
+    // not run on Android.
+    version (CRuntime_Bionic) {} else
```

Comment:
As the author of this test (and the test below) I'm now wondering if it might have been a mistake to rely on implementation defined behavior here (actually undefined behavior according to the [Posix standard](http://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_destroy.html)). I assumed that [`pthread_mutex_trylock`](http://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_mutex_trylock.html) would return `EINVAL` on all platforms if called with a previously destroyed mutex (otherwise what would be the point of specifying such return value in the standard?).

I would love to find a better way to do this, but I can't think of any other alternatives at the moment.